webapp/controller/PageOverview.controller.js

```diff
@@ -0,0 +1,199 @@
+sap.ui.define([
+	"sap/ui/core/mvc/Controller",
+	'sap/ui/model/json/JSONModel',
+	"sap/ui/model/Filter",
+	"sap/ui/model/FilterOperator",
+	'sap/m/Token',
+], function (Controller, JSONModel, Filter, FilterOperator, Token) {
+	"use strict";
+
+	return Controller.extend("products.app.controller.PageOverview", {
+		
+		onInit: function() {
+			const oModel = new JSONModel();
+			oModel.loadData("../model/products.json");
+		
+			this.getView().setModel(oModel);
+		},
+
+		onAfterRendering: function() {
+			const oMultiInput = this.byId("multiInput");
+			oMultiInput.attachTokenUpdate((event) => {
+				if (event.getParameter("type") === "removed") {
+					const oModel = this.getView().getModel();
+					const aSuppliers = oModel.getProperty("/Suppliers");
+					
+					const aRemovedTokens = event.getParameter("removedTokens");
+				
+					const aRemainingTokens = oMultiInput.getTokens().filter(function (token) {
+					  return !aRemovedTokens.includes(token);
+					});
+					
+					const aSelectedTokens = aRemainingTokens.map(el => el.getText());
+					const aSelectedId = aSuppliers
+											.filter((el) => aSelectedTokens.includes(el.SuppliersName))
+											.map((el) => el.SupplierId);
+					
+					this.onFilter(aSelectedId);
+				}
+			})
+		},
+
+		getSuppliersName: function(data) {
+			const oModel = this.getView().getModel();
+			const aSuppliers = oModel.getProperty("/Suppliers");			
+			
+			return aSuppliers
+					.filter((el) => data.includes(el.SupplierId))
+					.map(el => el.SuppliersName)
+					.join(", ");
+		},
+
+		getCombinedFilter: function(aSelectedId) {
+			const sQuerySearch = this.byId("searcher").getValue();
+			const sSelectedKey = this.byId("availabilitySelect").getSelectedKey();
+			
+			const sAvailability = "all";
+			const oAvailabilityFilter = new Filter(
+				"Availability", 
+				FilterOperator.NE,
+				sSelectedKey === sAvailability ? sSelectedKey : JSON.parse(sSelectedKey.toLowerCase())
```

Comment:
1. Do you really need to JSON.parse here?
2. sSelectedKey means the value is of type string. If it is not string, please change the notation (oSelectedKey, mSelectedKey etc). But I believe the value is of type string and you don't need any parsing here

Reply:
I need JSON.parse(sSelectedKey.toLowerCase()) because it help me parse string like "false" and "true" in boolean false and true. I just found this in internet). Better to replace this?